keps/sig-auth/4872-harden-kubelet-cert-validation/README.md

```diff
@@ -0,0 +1,390 @@
+
+The average `apiserver_request_duration_seconds` for logs/exec/port-forward requests is within reasonable limits.
+A rising value after enabling this feature could signal overhead introduced by the extra validation.
+
+In addition, the number of TLS connections made from API server to nodes should not increase.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `kube_apiserver_validation_kubelet_cert_cn_total`
+  - Components exposing the metric: kube-apiserver
+  - If the feature is enabled, and the metric increases for the `failure` label, it signals a problem.
+  - If the service is healthy, the metric should increase.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+We could add a metric to track the time spent per request on the CN validation.
+
+However, we consider this metric to not provide enough value to justify the work to maintain it.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No. This only affects streaming APIs and these are not covered by SLIs/SLOs.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+It's part of the API server, so the feature will be unavailable.
+
+###### What are other known failure modes?
+
+- [API server can't connect to Nodes with custom kubelet serving certificates that don't follow the `system:node:<node-name>` convention]
+  - Detection: `kubectl logs` returns a certificate validation error. 
+  - Mitigations: disable the validation by not setting `--enable-kubelet-cert-cn-validation` flag.
+  - Diagnostics: error is returned by the API server, no additional logging needed.
+  - Testing: We will have tests for this, this is basically testing that the feature works. 
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+* Implementation options discussion: https://docs.google.com/document/d/1RqhAkGov_coHsB3lbAo-qfQl1MOfYvgpPUjiGMJ_3PY
```

Comment:
Nit: add the information about filling in this document.

Reply:
@soltysh
I updated it, let me know if that's what you meant 🙏🏻

Reply:
What @g-gaston put there is exactly what I was looking for, thank you!